pkg/target/kusttarget.go

```diff
@@ -228,11 +228,6 @@ func (kt *KustTarget) AccumulateTarget() (
 		return nil, errors.Wrapf(
 			err, "merging config %v", tConfig)
 	}
-	err = ra.MergeVars(kt.kustomization.Vars)
-	if err != nil {
-		return nil, errors.Wrapf(
-			err, "merging vars %v", kt.kustomization.Vars)
-	}
```

Comment:
need transform and generation before merge var otherwise some resources referenced by var will not exist.

Reply:
MergeVars merely accumulates Var definitions as we read kustomization files; the Vars aren't used until ResolveVars is called.  There are no checks for references.  There are only checks for uniqueness.

ResolveVars isn't called until after all generators and transformers have run.

If this call is not moved, do existing tests fail?

I'm curious what fails.  What about my understanding is failing here?

Reply:
Yes, if we do not move, the existing test will fail. For instance, if you have a configuration generator to generate configmap, and a variable reference this variable, MergeVar will fail because in this pr, we try to find the resource referenced by var in MergeVars, and since configmap is not generated yet, it fails to find the resources.

Reply:
Ah thanks, I appreciate that!